src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,248 @@
+package edu.cornell.kfs.module.purap.rest.resource;
+
+import com.google.gson.Gson;
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.module.purap.dataaccess.CuEinvoiceDao;
+import edu.cornell.kfs.module.purap.dataaccess.impl.CuEinvoiceDaoOjb;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.kuali.kfs.krad.dao.LookupDao;
+import org.kuali.kfs.krad.util.KRADPropertyConstants;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.purap.PurapPropertyConstants;
+import org.kuali.kfs.module.purap.businessobject.PurApItem;
+import org.kuali.kfs.module.purap.document.PurchaseOrderDocument;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@Path("api")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class CuEinvoiceApiResource {
+
+    private static final Log LOG = LogFactory.getLog(CuEinvoiceApiResource.class);
+
+    private LookupDao lookupDao;
+    private CuEinvoiceDao cuEinvoiceDao;
+    private Gson gson = new Gson();
+
+    @Context
+    protected HttpServletRequest servletRequest;
+
+    @Context
+    protected HttpServletResponse servletResponse;
+
+    @GET
+    public Response describeEinvoiceApiResource() {
+        return Response.ok(CUPurapConstants.Einvoice.EINVOICE_KFS_API_DESCRIPTION).build();
+    }
+
+    @GET
+    @Path("vendors")
+    public Response getVendor(@Context HttpHeaders headers) {
+        try {
+            List<String> vendorNumbers = getVendorNumbersFromRequest();
+            List<VendorDetail> vendors = getCuEinvoiceDao().getVendors(vendorNumbers);
+            List<Properties> vendorsSerialized = vendors.stream().map(vendor -> getVendorProperties(vendor)).collect(Collectors.toList());
+            return Response.ok(gson.toJson(vendorsSerialized)).build();
+        } catch (BadRequestException ex) {
```

Comment:
I think . we could eliminate the catch for BadRequestException, and just used the generic exception catch.  Unless there is a reason to NOT log the BadRequestException case.

Reply:
I have added logging statements to the catch block of the BadRequestException. For the generic Exception the server is responding with status code 500 internal server error, but with BadRequestException it should be responding with status code 400 Bad Request

Reply:
Thanks for the clarification